### PR TITLE
fix: authentication allowing any password

### PR DESCRIPTION
**Pull request by G3root**
currently the login process allows any random password as a legit password, this PR aims to fix that.

### CODE REVIEW

File: packages/lib/next-auth/error-codes.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export const ErrorCodes = {
```
Comment (G3root) [inline packages/lib/next-auth/error-codes.ts:1]: Error code to show an banner on the login form if an error occurs

File: packages/lib/next-auth/auth-options.ts [hunk 2]
```diff
@@ -23,21 +24,23 @@ export const NEXT_AUTH_OPTIONS: AuthOptions = {
...
+        });
 
-        if (!user || !user.password) {
-          return null;
+        if (!user.password) {
+          throw new Error(ErrorCodes.UserMissingPassword);
         }
 
-        const isPasswordsSame = compare(password, user.password);
+        const isPasswordsSame = await compare(password, user.password);
```
Comment (G3root) [inline packages/lib/next-auth/auth-options.ts:40]: this seems to be the issue

Comment (Mythie) [reply packages/lib/next-auth/auth-options.ts:40]: I need to add an eslint rule for this 😓

Comment (adithyaakrishna) [reply packages/lib/next-auth/auth-options.ts:40]: If we have a `no-floating-promises`rule

Comment (adithyaakrishna) [reply packages/lib/next-auth/auth-options.ts:40]: We can combine it with `require-await` rule as well

**Review by Mythie — APPROVED** [review]

**Comment by Mythie** [comment]
Thanks Nafees! We weren't able to get your first PR in but this one is great 🚀 

thanks-adventure-time

**Pull request merged**